src/EnergyPlus/PondGroundHeatExchanger.cc

```diff
@@ -498,12 +497,12 @@ namespace PondGroundHeatExchanger {
 
         Real64 PondMass = this->Depth * this->Area *
                           FluidProperties::GetDensityGlycol(
-                              fluidNameWater, max(this->PondTemp, DataPrecisionGlobals::constant_zero), this->WaterIndex, RoutineName);
+                              "WATER", max(this->PondTemp, DataPrecisionGlobals::constant_zero), this->WaterIndex, RoutineName);
```

Comment:
@Myoldmopar I just saw this message.

The "correct" answer is `static constexpr std::string_view` for a compile-time known string that you want to give a name to. This is only available in C++17.

The next best option is `static constexpr fmt::string_view`. Simply because we have access to it. I think I'm going to move to this for my format strings soon.

`static const std::string` is the wrong answer for anything that's a "small" string (15 characters or less). This is going to add overhead in accessing the string object and entering the scope the variable is in (this overhead is checking to see if the object has been initialized or not).

For longer strings `static const std::string` is OK, as it would save a dynamic allocation on each function call / loop / whatever.

As an aside, in my personal code I've relying on using `auto` more and more for anything that's returned from a function. And the reasoning is simple: I want to avoid conversions. This is true regardless of if it is a `&` or a value type. Both can accidentally invoke implicit conversions, and even integral conversions (like `size_t` -> `int`) can add overhead.

If we had `-Wconversion` enabled then we would at least be aware of these conversions when they happen if we stick with explicit types.

(source: 3 years of teaching Object Lifetime classes and answering all of the questions students ask about all the things.)